src/lightning/pytorch/plugins/precision/amp.py

```diff
@@ -40,6 +40,9 @@ def __init__(
         device: str,
         scaler: Optional[torch.cuda.amp.GradScaler] = None,
     ) -> None:
+        if precision not in ["16-mixed", "bf16-mixed"]:
+            raise MisconfigurationException("`precision` must be '16-mixed' or 'bf16-mixed'")
```

Comment:
It's important that error messages include what they were given, not just what they expected

```suggestion
        if precision not in ("16-mixed", "bf16-mixed"):
            raise ValueError("`{type(self).__name__}(precision={precision!r})` must be '16-mixed' or 'bf16-mixed'")
```

Reply:
Done